frontend/src/pages/QueryBuilder/QueryBuilder.tsx

```diff
@@ -173,17 +173,29 @@ const QueryBuilder: FunctionComponent<QueryBuilderProps> = (props) => {
           data,
         })
         .then((response: AxiosResponse<Operation>) => {
-          if (response.status === 200 || response.status === 201) {
-            props.actions.operationSaved(true);
-            if (preview) {
-              props.history.push(`/queries/data/${response.data.id}/`);
-            } else {
-              props.history.push('/');
+          if (response.data.alias_creation_status !== 'p') {
```

Comment:
This shouldn't be here ... it's not possible that this could happen since if the code runs without interruption, it can only have two status ... `d` & `e` (probably only `d`)... So in addition to the error check on line 191, this check should also be done in `setActiveOperationByID` (I think), to be run when the user first opens the query builder to edit an existing operation. i.e. `if (operation.alias_creation_status !== 'd')`. Alternatively, u can have a `useEffect` that watches for changes in `props.activeOperation`, then checks the alias creation status.

Reply:
Oh yes,this sounds more practical